### DISCUSSION TITLE
Plans for Firebird RDBMS support?

### DISCUSSION THREAD

**Discussion by mikini** [TPC-Council/HammerDB#57] [Q&A] [answered]
I'm looking into evaluating database perfomance in different hosting scenarios involving the [Firebird RDBMS](https://en.wikipedia.org/wiki/Firebird_(database_server)). I see that [Firebird is not officially supported](https://www.hammerdb.com/docs/ch01s02.html) but I am wondering whether any work has been put into supporting Firebird, or there are any plans to support it at some point?

The [Firebird project itself](https://firebirdsql.org/) has [produced an oltp-emul tool](https://github.com/FirebirdSQL/oltp-emul/) which apparently performs some or all of the TPC-C specified tests. I'm trying to go with this but it is not easy to comprehend as a DB performance/TPC-C rookie. Also the tools seems somewhat unpolished and also riddled by the developer being Russian.

**Comment by sm-shaw** [comment, marked as answer]
One of the key motivations of the move to the TPC-Council is to open up HammerDB development to a wider community who wish to add support for additional databases and workloads and therefore if you wish to lead the initiative to add Firebird RDBMS support then this is more than welcome.  The structure of HammerDB is modular and therefore a new database can be added by adding a config file and the code under a directory of the database name.

**Comment by sm-shaw** [comment]
Closed as no further updates received.  Help can be requested for the method to add additional database support.

**Comment by mikini** [comment]
Sorry for not responding timely, I got catapulted away from this task by some other very high priority stuff. I think at the moment we have achieved the benchmarking results we needed. Would have been nice studying tcl and TPC adding firebird support to HammerDB but sadly I'm not in a position where I can take on that task.
